internal/gps/pkgtree/pkgtree.go

```diff
@@ -189,7 +199,15 @@ func ListPackages(fileRoot, importRoot string) (PackageTree, error) {
 		return PackageTree{}, err
 	}
 
-	return ptree, nil
+	return ptree, checkError(ptree)
```

Comment:
i don't think this is what we're looking for in an error case here. errors returned from `ListPackages()` itself are used to indicate that the parsing & treebuilding process itself encountered an error, not that there are problems in the data the tree is itself describing.

also, this violates the general principle that we should only ever return a real value XOR an error from a function.